recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -71,6 +71,18 @@ def _min_cppstd(self):
             return 14
         return 11
 
+    @property
+    def _default_cppstd(self):
+        if self.options.with_stl == "default":
+            return default_cppstd(self).replace("gnu", "")
```

Comment:
It seems this doesn’t match the `compiler.cppstd` setting.

Reply:
Please explain.

Reply:
`default_cppstd()` returns some hardcoded default cppstd value based on the compiler version. See:

https://github.com/conan-io/conan/blob/51731ef38732b07a0b17f1523790b71d4d24839e/conan/tools/build/cppstd.py#L68

Which eventually calls:

https://github.com/conan-io/conan/blob/51731ef38732b07a0b17f1523790b71d4d24839e/conan/internal/api/detect_api.py#L299

Reply:
Now I'm returning `compiler.cppstd` if set in profile.

Reply:
Should I remove the "default" value and return a Version specific value?

```
@property
    def _default_cppstd(self):
        if self.settings.compiler.cppstd:
            return str(self.settings.compiler.cppstd).replace("gnu", "")
        else:
            return default_cppstd(self).replace("gnu", "")

    @property
    def _default_stl_version(self):
        if self.options.with_stl and Version(self.version) >= "1.12":
            return f"CXX{self._default_cppstd}"
        return self.options.with_stl
```

Reply:
@cynix whats your opinion?

Reply:
Do you mean just make the option True/False, and return the cppstd value when set to True? I think that’s fine too :)